llama_index/vector_stores/qdrant.py

```diff
@@ -347,16 +420,60 @@ def query(
         Args:
             query (VectorStoreQuery): query
         """
+        from qdrant_client import models as rest
         from qdrant_client.http.models import Filter
 
         query_embedding = cast(List[float], query.query_embedding)
+        query_filter = cast(Filter, self._build_query_filter(query))
 
-        response = self._client.search(
-            collection_name=self.collection_name,
-            query_vector=query_embedding,
-            limit=cast(int, query.similarity_top_k),
-            query_filter=cast(Filter, self._build_query_filter(query)),
-        )
+        if query.mode == VectorStoreQueryMode.HYBRID and not self.enable_hybrid:
+            raise ValueError(
+                "Hybrid search is not enabled. Please build the query with "
+                "`enable_hybrid=True` in the constructor."
+            )
+        elif query.mode == VectorStoreQueryMode.HYBRID and self.enable_hybrid:
+            sparse_indices, sparse_embedding = self._sparse_query_fn(
+                [query.query_str],
+            )
+            sparse_top_k = query.sparse_top_k or query.similarity_top_k
+
+            sparse_response = self._client.search_batch(
+                collection_name=self.collection_name,
+                requests=[
+                    rest.SearchRequest(
+                        vector=rest.NamedVector(
+                            name="text-dense",
+                            vector=query_embedding,
+                        ),
+                        limit=query.similarity_top_k,
+                        filter=query_filter,
+                        with_payload=True,
+                    ),
+                    rest.SearchRequest(
+                        vector=rest.NamedSparseVector(
+                            name="text-sparse",
+                            vector=rest.SparseVector(
+                                indices=sparse_indices[0],
```

Comment:
what is this line for. indicate the indice to query on? so we could have multiple sparse indices?

Reply:
Sparse vectors are made up of indicies and values

Normally, the vector is like 30,000 dimensions

But we reduce to only the non-zero values, and keep track of the indexes of those non-zero values.

This is what qdrant expects, as well as pinecone too actually